src/AnalyzerInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dhii\Services\Tools;
+
+use Dhii\Services\ServiceInterface;
+
+interface AnalyzerInterface
+{
+    /**
+     * Analyzes a set of factories and extensions.
+     *
+     * @param array<callable|ServiceInterface> $factories  The list of factories to analyze.
+     * @param array<callable|ServiceInterface> $extensions The list of extensions to analyze.
```

Comment:
Can these be `iterable` instead of `array`?

Reply:
Not sure. It's been a while since I wrote this. I'll get back to you on this.

Reply:
They can't be `iterable` because the desired type is, in actuality, `array<string, callable|ServiceInterface>`.